h/static/scripts/filter/document-title.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+module.exports = [function() {
+  /**
+   * Return a nice displayable string representation of a document's title.
+   *
+   * @returns {String} The document's title preceded on "on " and hyperlinked
+   *   to the document's URI. If the document has no http(s) URI then don't
+   *   hyperlink the title. If the document has no title then return ''.
+   *
+   */
+  function documentTitleFilter(document) {
+    var title = document.title;
+    var uri = document.uri;
+    if (uri && !(uri.startsWith('http://') || uri.startsWith('https://'))) {
+      // We only link to http(s) URLs.
+      uri = null;
+    }
+
+    if (title && uri) {
+      return ('on &ldquo;<a target="_blank" href="' + uri + '">' + title +
```

Comment:
Ok I've pushed a commit that fixes this (the tests will be broken, I'll fix them).

This does make me think two things though:

1. We should be escaping HTML in annotation fields such as title server-side
2. This filters approach smells - we shouldn't be creating HTML by concatenating strings. On the other hand, I wanted to make the logic testable. Maybe @tilgovi is right that `AnnotationController` should just add some helpful booleans like `isLinkable` to make the logic in the templates really simple to write.